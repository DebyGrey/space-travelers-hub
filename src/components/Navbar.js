import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <>
    <div className="navbar">
      <div className="nav-left">
        <h1>Space Travelers&apos; Hub</h1>
        <nav>
          <ul>
            <li>
              <NavLink to="/">Rockets</NavLink>
            </li>
            <li>
              <NavLink to="missions">Missions</NavLink>
            </li>
            |
            <li>
              <NavLink to="my-profile">My Profile</NavLink>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </>
);
export default Navbar;
