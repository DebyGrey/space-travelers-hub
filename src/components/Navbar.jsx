import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <>
    <div className="navbar border">
      <div className="nav-left">
        <h1>Space Travelers&apos; Hub</h1>
        <nav>
          <ul>
            <li>
              <NavLink to="/">Rockets</NavLink>
            </li>
            <li>
              <NavLink to="missions">Missions</NavLink>
            </li>
            |
            <li>
              <NavLink to="my-profile">My Profile</NavLink>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </>
);
export default Navbar;
