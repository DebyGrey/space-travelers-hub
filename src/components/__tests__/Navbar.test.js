import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  test('renders the logo', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const logoElement = screen.getByAltText('logo'); // Assuming the Logo component uses an <img> tag with the alt text 'Logo'
    expect(logoElement).toBeInTheDocument();
  });

  test('renders the navigation menu items', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const menuItems = screen.getAllByRole('link');
    const expectedUrls = [
      { id: 1, name: "Rockets", path: "/" },
      { id: 2, name: "Missions", path: "/missions" },
      { id: 3, name: "My Profile", path: "/my-profile" },
    ];

    expect(menuItems).toHaveLength(expectedUrls.length);

    menuItems.forEach((menuItem, index) => {
      expect(menuItem).toHaveTextContent(expectedUrls[index].name);
      expect(menuItem).toHaveAttribute(
        'href',
        expectedUrls[index].path,
      );
    });
  });

  describe('Navbar', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
