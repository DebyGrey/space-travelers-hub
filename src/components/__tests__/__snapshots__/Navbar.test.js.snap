// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Navbar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-[3.2rem] font-manrope"
  >
    <div
      class="border-b flex justify-between items-center px-[1.8rem] py-[1.5rem]"
    >
      <div
        class="flex justify-start items-center gap-x-4"
      >
        <img
          alt="logo"
          class="h-[60px] "
          src="logo.png"
        />
        <h1
          class="text-3xl font-bold"
        >
          Space Travelers' Hub
        </h1>
      </div>
      <nav
        class="nav-left "
      >
        <ul
          class="flex justify-end items-center"
        >
          <li
            class="last:border-l h-auto py-0 border-gray-400 px-2"
          >
            <a
              aria-current="page"
              class="text-blue-600 underline"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li
            class="last:border-l h-auto py-0 border-gray-400 px-2"
          >
            <a
              class="text-blue-600"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="last:border-l h-auto py-0 border-gray-400 px-2"
          >
            <a
              class="text-blue-600"
              href="/my-profile"
            >
              My-profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
